docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,137 @@
+}
+
+type BrokerSpec struct {
+  // Empty for now
+}
+
+type BrokerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	Address duckv1alpha1.Addressable `json:"address,omitempty"`
+}
+```
+
+### Trigger
+
+A Trigger represents a registration of interest (request for delivery to an
+Addressable object or URL) in a filtered selection of events delivered to a
+Broker. For the MVP, Triggers may only target Addressable objects in the same
+namespace.
+
+#### Buffering
+
+Triggers are assumed to buffer (durably queue) filtered events from the Broker
+from the time they are created, and to deliver the buffered events to the target
+Addressable as it is able to receive them.
+
+#### Filtering
+
+An event consumer which is interested in several sets of events may need to
+create multiple Triggers, each with an ANDed set of filter conditions which
+select the events. In the case of multiple Triggers with the same target whose
+filter conditions select an overlapping set of events, events will be delivered
+once for each Trigger which selects the event (i.e. Trigger selection criteria
+are independent, not aggregated in the Broker).
+
+At some later point, we expect that a more complex set of filtering expressions
+will be supported. For the initial MVP, exact match on the CloudEvents
+[`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type) and
+[`source`](https://github.com/cloudevents/spec/blob/master/spec.md#source)
+attributes will be supported.
+
+#### Trigger Object Definition
+
+```golang
+type Trigger struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec TriggerSpec `json:"spec,omitempty"`
+	Status TriggerStatus `json:"status,omitempty"`
+}
+
+type TriggerSpec struct {
+	// Defaults to 'default'.
+	Broker string `json:"broker,omitempty"`
+  Filter *TriggerFilter `json:"filter,omitempty"`
+  // As from Subscription's `spec.subscriber`
+	Target *SubscriberSpec `json:"target,omitempty"`
+}
+
+type TriggerFilter struct {
+  // Exact match expressions; if empty, all values are matched.
+	Type   string `json:"type,omitempty"`
+	Source string `json:"source,omitempty"`
+}
+
+type TriggerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	SubscriberURI string `json:"subscriberURI,omitempty"`
+}
+```
+
+## Backing documents
+
+[2019-02-22 Decision Making -- UX targets](https://docs.google.com/spreadsheets/d/16aOhfRnkaGcQIOR5kiumld-GmrgGBIm9fppvAXx3mgc/edit)
+
+Accepted decisions (2+ votes)
+
+- Have a "Broker" object
+  - Default sink on sources
+  - Enable with namespace/default Broker
+- Have a "Trigger" object
+  - "filter" subfield (not top-level filters)
+  - "filter" has exact type match
+  - "filter" has exact source match
+  - target is same-namespace
+- Registry requirements are not clear enough to include in MVP
```

Comment:
Need clarification. Does this mean
a) Registry is not in scope for MVP? 
OR
b) Registry is in scope for MVP but not yet specd out?

My understanding is that MVP scope is defined from a end customer perspective. - What is the minimal experience that is needed to get a meaningful feedback?
The scope is being discussed in the user stories document, the spec and detail requirements could be discussed in the UX proposal doc.

Reply:
There was no clear consensus as to how the registry should be implemented or what functionality was needed; my assumption is that our MVP includes only functionality we can describe and implement.

I would assume that clear user stories which compellingly motivated the registry would also lend clarity to whether the Registry is in scope for MVP. Absent such evidence, I must conclude that it is not.

Reply:
I understand that there was no clear consensus on registry design. However the bigger question is scope of MVP. 
As per the [user stories](https://docs.google.com/document/d/15uhyqQvaomxRX2u8s0i6CNhA86BQTNztkdsLUnPmvv4/edit) doc we have discussed, registry is seems to be in scope for MVP.
Given the current timelines we are targeting my suggestion will be to start with registry design and include an experimental version of MVP in 0.5 which will be enough for end users to play with, knowing that it is an experimental feature. This way we can have a complete experience shipped in 0.5, blog about it, and get feedback which will help us design registry incrementally.

Reply:
Removed the phrase "MVP" in favor of the slightly wordier "this decision". Does that help?